webapp-client/src/component/AddCategoryForm.js

```diff
@@ -1,32 +1,73 @@
 import React from 'react';
+import ErrorMessage from './ErrorMessage';
 
 export default class AddCategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasError: false,
+    };
+    this.createCategory = this.createCategory.bind(this);
+    this.clearHasError = this.clearHasError.bind(this);
+  }
   // This function will be changed later to directly POST request to db instead of calling addCategory
-  createCategory = (e) => {
+  async createCategory(e) {
     e.preventDefault();
     const category = e.target.elements['add-category'].value;
```

Comment:
Should we add a check to only send the fetch if the category is not an empty string?